### PR TITLE
Test reporter

### PR DESCRIPTION
**Pull request by dnfield**
This PR wraps all test output so that errors are printed last.  It also sends all test result data to a bigquery table we can look at to get better info about what tests failed and when without needing to visit specific builder pages.

It's only implemented for Cirrus at the moment.  It should be possible to implement for LUCI as well, but I'd rather tackle that separately.

### CODE REVIEW

**Comment by dnfield** [comment]
This is working on Cirrus, need to check into how I can make it work for LUCI.

**Comment by dnfield** [comment]
This is still TODO on LUCI but would be an improvement for Cirrus.  Marking ready for review.

/cc @goderbauer

File: dev/bots/flutter_compact_formatter.dart [hunk 1]
```diff
@@ -0,0 +1,259 @@
+import 'dart:convert';
```
Comment (gspencergoog) [inline dev/bots/flutter_compact_formatter.dart]: Copyright?

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Very cool.  I can't wait to see the data analysis we can do with this...

**Comment by jonahwilliams** [comment]
This doesn't appear to be working properly: https://cirrus-ci.com/task/4970475997364224

**Comment by dnfield** [comment]
Something's messed up on Windows.  Checking.

**Comment by dnfield** [comment]
That will be fixed by #28970

File: dev/bots/test.dart [hunk 2]
```diff
@@ -140,13 +144,31 @@ Future<void> _runSmokeTests() async {
   await _verifyVersion(path.join(flutterRoot, 'version'));
 }
 
+Future<bq.BigqueryApi> _getBigqueryApi() async {
+  // TODO(dnfield): How will we do this on LUCI?
```
Comment (Hixie) [inline dev/bots/test.dart]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-checking-in-comments-that-ask-questions

File: dev/bots/test.dart [hunk 3]
```diff
@@ -140,13 +144,31 @@ Future<void> _runSmokeTests() async {
...
+  final String privateKey = Platform.environment['GCLOUD_SERVICE_ACCOUNT_KEY'];
+  if (privateKey == null || privateKey.isEmpty) {
+    return null;
+  }
+  final auth.ServiceAccountCredentials accountCredentials = auth.ServiceAccountCredentials( //.fromJson(credentials);
```
Comment (Hixie) [inline dev/bots/test.dart]: nit: stray comment on this line?

**Comment by Hixie** [comment]
What's the background here?

**Comment by dnfield** [comment]
The compact formatter makes it really hard to find what tests actually failed - particularly since Cirrus only shows us the tail of the log (which almost never contains the test that failed).

I also would like to see all of our test results in a database so we could find failing tests rather than checking failing bots.

This is currently turned off though, as it had a couple problems I haven't been able to fully sort through yet.

**Pull request merged**